flake.nix

```diff
@@ -2,29 +2,30 @@
   inputs.nixpkgs.url = "github:nixos/nixpkgs";
   inputs.flake-utils.url = "github:numtide/flake-utils";
 
-  outputs = { self, nixpkgs, flake-utils, ... }@inputs:
+  outputs = { nixpkgs, flake-utils, ... }:
     flake-utils.lib.eachDefaultSystem (system:
       let
         pkgs = import nixpkgs { inherit system; };
-        shellPackages = [
-          "jre"
-          "sbt"
-          "nodejs-16_x"
-          "yarn"
+        shellPackages = with pkgs; [
+          temurin-jre-bin-17
+          nodejs-18_x
+          yarn
+          (pkgs.sbt.override { jre = pkgs.temurin-jre-bin-17; })
```

Comment:
**thought:** I had to override the `jre` version used by this particular derivation of `sbt` (which is still 1.8.x version) as it was using too recent version of `jre` - 21.

I then decided to completely drop the idea of `shellInputs` being just a list of strings since this required the packages to be defined in two places without any particular benefit. Also the errors from nix evaluator were harder to debug when using strings.